### PR TITLE
Dependabots/26 nov 2024

### PR DESCRIPTION
**Pull request by alexjavabraz**
November dependabot alerts added here

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM node:18-alpine as build
+FROM node:23-alpine as build
```
Comment (lserra-iov) [inline Dockerfile:1]: Are we sure about using a non LTS version here?

File: .github/workflows/codeql.yml [hunk 2]
```diff
@@ -26,18 +26,18 @@ jobs:
 
     steps:
       - name: Checkout
-        uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871 # v4.2.1
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
 
       - name: Initialize CodeQL
-        uses: github/codeql-action/init@5973e2313ac1b9aaedb1f911a1e91228cde07d28 # v2.26.0
+        uses: github/codeql-action/init@f09c1c0a94de965c15400f5634aa42fac8fb8f88 # v3.27.5
```
Comment (annipi) [inline .github/workflows/codeql.yml:32]:
```suggestion
        uses: github/codeql-action/init@aa578102511db1f4524ed59b8cc2bae4f6e88195 # v3.27.6
```

File: .github/workflows/codeql.yml [hunk 3]
```diff
@@ -26,18 +26,18 @@ jobs:
...
         with:
           languages: ${{ matrix.language }}
           queries: +security-and-quality
 
       - name: Autobuild
-        uses: github/codeql-action/autobuild@5973e2313ac1b9aaedb1f911a1e91228cde07d28 # v2.26.0
+        uses: github/codeql-action/autobuild@f09c1c0a94de965c15400f5634aa42fac8fb8f88 # v3.27.5
```
Comment (annipi) [inline .github/workflows/codeql.yml:38]:
```suggestion
        uses: github/codeql-action/autobuild@aa578102511db1f4524ed59b8cc2bae4f6e88195# v3.27.6
```

File: .github/workflows/codeql.yml [hunk 4]
```diff
@@ -26,18 +26,18 @@ jobs:
...
 
       - name: Perform CodeQL Analysis
-        uses: github/codeql-action/analyze@5973e2313ac1b9aaedb1f911a1e91228cde07d28 # v2.26.0
+        uses: github/codeql-action/analyze@f09c1c0a94de965c15400f5634aa42fac8fb8f88 # v3.27.5
```
Comment (annipi) [inline .github/workflows/codeql.yml:41]:
```suggestion
        uses: github/codeql-action/analyze@aa578102511db1f4524ed59b8cc2bae4f6e88195 # v3.27.6
```

**Review by annipi — APPROVED** [review]

File: .github/workflows/codeql.yml [hunk 5]
```diff
@@ -26,18 +26,18 @@ jobs:
...
       - name: Initialize CodeQL
-        uses: github/codeql-action/init@5973e2313ac1b9aaedb1f911a1e91228cde07d28 # v2.26.0
+        uses: github/codeql-action/init@aa578102511db1f4524ed59b8cc2bae4f6e88195 # v3.27.6
         with:
           languages: ${{ matrix.language }}
           queries: +security-and-quality
 
       - name: Autobuild
-        uses: github/codeql-action/autobuild@5973e2313ac1b9aaedb1f911a1e91228cde07d28 # v2.26.0
+        uses: github/codeql-action/autobuild@aa578102511db1f4524ed59b8cc2bae4f6e88195# v3.27.6
```
Comment (annipi) [inline .github/workflows/codeql.yml:38]:
```suggestion
        uses: github/codeql-action/autobuild@aa578102511db1f4524ed59b8cc2bae4f6e88195 # v3.27.6
```

**Pull request merged**